src/eos-updater-fetch.c

```diff
@@ -744,9 +758,18 @@ handle_fetch (EosUpdater            *updater,
       return TRUE;
     }
 
+  fetch_data = g_new0 (FetchData, 1);
+  fetch_data->update_id = g_strdup (eos_updater_get_update_id (updater));
+  fetch_data->update_refspec = g_strdup (eos_updater_get_update_refspec (updater));
+  fetch_data->progress = ostree_async_progress_new_and_connect (update_progress, updater);
```

Comment:
This is a very minor nit, but I've found that for closures like this, it usually makes sense to have a separate constructor func, eg `fetch_data_new` which takes all the parameters it needs to construct the closure and then takes ownership of them through `g_strdup`, `g_object_ref` etc there.

That way if you ever need to construct `FetchData` in more than one place, changing ownership semantics of its members implies that only a single function needs to be updated.

Reply:
I find it’s clearer to have the code inline in the first instance. If we end up needing to create a `FetchData` from a second call site, then it’s time to factor the construction out.